### PR TITLE
Address golint warnings for package cmd, control, core and mgmt

### PR DESCRIPTION
**Pull request by candysmurf**

### CODE REVIEW

**Comment by tiffanyfj** [comment]
Seems you're having the same issue I have been having lately--you need to get gofmt working to properly format your code so it doesn't complain.

File: cmd/snapctl/task.go [hunk 1]
```diff
@@ -48,7 +48,9 @@ var (
 	unionParseFormat = timeParseFormat + " " + dateParseFormat
 )
 
+// const defines constants for the length of a string.
 const (
+	// K defines the length of a string.
```
Comment (danielscottt) [inline cmd/snapctl/task.go]: K-G are not for string length, I don't believe.  They are used to truncate task hit and miss counts.

File: cmd/snapctl/task.go [hunk 2]
```diff
@@ -48,7 +48,9 @@ var (
 	unionParseFormat = timeParseFormat + " " + dateParseFormat
 )
 
+// const defines constants for the length of a string.
```
Comment (danielscottt) [inline cmd/snapctl/task.go]: Don't think this one is necessary.

File: control/control.go [hunk 3]
```diff
@@ -47,6 +47,7 @@ import (
 	"github.com/intelsdi-x/snap/pkg/psigning"
 )
 
+// Plugin tokens generated by plugin and passed to control
```
Comment (danielscottt) [inline control/control.go]: These are enums describing the trust (plugin signing) settings.

File: control/runner.go [hunk 4]
```diff
@@ -239,7 +249,7 @@ func (r *runner) stopPlugin(reason string, ap *availablePlugin) error {
 	return nil
 }
 
-// Empty handler acting as placeholder until implementation. This helps tests
+// HandleGomitEvent is an empty handler acting as placeholder until implementation that helps tests
```
Comment (danielscottt) [inline control/runner.go]: Not empty any more.  This comment is old :(

File: core/metric.go [hunk 5]
```diff
@@ -27,6 +27,7 @@ import (
 	"github.com/intelsdi-x/snap/core/cdata"
 )
 
+// Label struct type
```
Comment (danielscottt) [inline core/metric.go]:
```go
// Label is used when a metric uses a wildcard in its namespace
```

File: core/metric.go [hunk 6]
```diff
@@ -50,12 +51,14 @@ type RequestedMetric interface {
 	Version() int
 }
 
+// CatalogedMetric interface
```
Comment (danielscottt) [inline core/metric.go]:
```go
// CatalogedMetric is the type returned when querying control's MetricCatalog
```

File: cmd/snapctl/task.go (hunk 1 above)
Comment (tiffanyfj) [inline cmd/snapctl/task.go]: K-G are metric prefixes (kilo, mega, giga)

File: core/metric.go [hunk 7]
```diff
@@ -50,12 +51,14 @@ type RequestedMetric interface {
...
 type CatalogedMetric interface {
 	RequestedMetric
 	LastAdvertisedTime() time.Time
 	Policy() *cpolicy.ConfigPolicyNode
 }
 
+// JoinNamespace joins metric namespaces
```
Comment (danielscottt) [inline core/metric.go]:
```go
// JoinNamespace concatenates a metric namespace
```

File: core/plugin.go [hunk 8]
```diff
@@ -28,14 +28,17 @@ import (
...
 )
 
+// Plugin interface
 type Plugin interface {
 	TypeName() string
 	Name() string
 	Version() int
 }
 
+// PluginType integer
```
Comment (danielscottt) [inline core/plugin.go]:
```go
// PluginType is used as an enum to differentiate types of snap plugins
```

File: core/plugin.go [hunk 9]
```diff
@@ -28,14 +28,17 @@ import (
...
 type PluginType int
 
+// ToPluginType transforms a string into PluginType
```
Comment (danielscottt) [inline core/plugin.go]:
```go
// ToPluginType transforms the name, e.g. "collector", to the corresponding enum
```

File: core/plugin.go [hunk 10]
```diff
@@ -57,13 +62,15 @@ func (pt PluginType) String() string {
 	}[pt]
 }
 
+// const List of plugin type
```
Comment (danielscottt) [inline core/plugin.go]: enum of plugin types

File: core/plugin.go [hunk 11]
```diff
@@ -57,13 +62,15 @@ func (pt PluginType) String() string {
...
 const (
-	// List of plugin type
 	CollectorPluginType PluginType = iota
 	ProcessorPluginType
 	PublisherPluginType
 )
 
+// AvailablePlugin the public interface for
```
Comment (danielscottt) [inline core/plugin.go]:
```go
// AvailablePlugin is the public interface for a running plugin
```

File: core/plugin.go [hunk 12]
```diff
@@ -71,9 +78,9 @@ type AvailablePlugin interface {
 	ID() uint32
 }
 
-// the public interface for a plugin
+// CatalogedPlugin the public interface for a plugin
 // this should be the contract for
-// how mgmt modules know a plugin
+// how mgmt modules know a plugin.
```
Comment (danielscottt) [inline core/plugin.go]: I'd lose the second 2 lines.

File: core/plugin.go [hunk 13]
```diff
@@ -82,21 +89,24 @@ type CatalogedPlugin interface {
 	LoadedTimestamp() *time.Time
 }
 
-// the collection of cataloged plugins used
+// PluginCatalog the collection of cataloged plugins used
```
Comment (danielscottt) [inline core/plugin.go]:
```go
PluginCatalog _is_ the collectrion of _loaded_ plugins
```

File: core/plugin.go [hunk 14]
```diff
@@ -82,21 +89,24 @@ type CatalogedPlugin interface {
...
 // by mgmt modules
 type PluginCatalog []CatalogedPlugin
 
+// SubscribedPlugin the public interface for subscribed plugins
```
Comment (danielscottt) [inline core/plugin.go]: SubscribedPlugin is used when ./scheduler calls SubscribeDeps on control.

File: core/plugin.go [hunk 15]
```diff
@@ -82,21 +89,24 @@ type CatalogedPlugin interface {
...
 type SubscribedPlugin interface {
 	Plugin
 	Config() *cdata.ConfigDataNode
 }
 
+// RequestedPlugin struct type
```
Comment (danielscottt) [inline core/plugin.go]: RequestedPlugin is used by theAPI when loading a plugin.

File: core/task.go [hunk 16]
```diff
@@ -29,8 +29,10 @@ import (
 	"github.com/intelsdi-x/snap/scheduler/wmap"
 )
 
+// TaskState integer type
```
Comment (danielscottt) [inline core/task.go]: TaskState is an enum of the possible states of a task.

File: core/task.go [hunk 17]
```diff
@@ -29,8 +29,10 @@ import (
...
 type TaskState int
 
+// const task states
```
Comment (danielscottt) [inline core/task.go]: Lose this line.

File: core/task.go [hunk 18]
```diff
@@ -29,8 +29,10 @@ import (
...
 const (
 	TaskDisabled TaskState = iota - 1
```
Comment (danielscottt) [inline core/task.go]: Task disabled due to consecutive failures

Comment (danielscottt) [inline core/task.go]: TaskDisabled is the state of a task when it has reached its consecutive failure limit.

File: core/task.go [hunk 19]
```diff
@@ -29,8 +29,10 @@ import (
...
 	TaskStopped
```
Comment (danielscottt) [inline core/task.go]: TaskStopped is a user-interactionable state.  A task is not _running.

File: core/task.go [hunk 20]
```diff
@@ -51,10 +54,13 @@ var (
 	}
 )
 
+// TaskWatcherCloser interface
```
Comment (danielscottt) [inline core/task.go]: Used to close an open watcher

File: core/task.go [hunk 21]
```diff
@@ -89,6 +97,7 @@ type Task interface {
 	Schedule() schedule.Schedule
 }
 
+// TaskOption data type
```
Comment (danielscottt) [inline core/task.go]: Used to provide optional parameters when constructing Tasks

File: core/workflow.go [hunk 22]
```diff
@@ -19,13 +19,16 @@ limitations under the License.
 
 package core
 
+// WorkflowState int type
```
Comment (danielscottt) [inline core/workflow.go]: Here and below at 31 can be changed to be similar to what was stated above re: Task states.

File: mgmt/rest/client/client.go [hunk 23]
```diff
@@ -39,13 +39,18 @@ import (
 )
 
 var (
+	// CompressUpload A flag indicates if an upload requires compressing
```
Comment (danielscottt) [inline mgmt/rest/client/client.go]: _compressed_.

File: mgmt/rest/client/client.go [hunk 24]
```diff
@@ -39,13 +39,18 @@ import (
...
+	// ErrUnknown The error message for unknown error
+	ErrUnknown = errors.New("Unknown error calling API")
+	// ErrNilResponse The error message for Nil response
 	ErrNilResponse = errors.New("Nil response from JSON unmarshalling")
-	ErrDirNotFile  = errors.New("Provided plugin path is a directory not file")
+	// ErrDirNotFile The error message for plugin path is a directory
+	ErrDirNotFile = errors.New("Provided plugin path is a directory not file")
 )
 
+// const A list of content type constants
```
Comment (danielscottt) [inline mgmt/rest/client/client.go]: Same enum comments.

File: mgmt/rest/client/client.go [hunk 25]
```diff
@@ -60,6 +65,7 @@ var (
 	}
 )
 
+// Client struct type
```
Comment (danielscottt) [inline mgmt/rest/client/client.go]: A REST API client for snap

File: mgmt/rest/client/metric.go [hunk 26]
```diff
@@ -27,7 +27,7 @@ import (
 )
 
 var (
-	// The default response error.
+	// ErrAPIResponseMetaType The default response error.
```
Comment (danielscottt) [inline mgmt/rest/client/metric.go]: Used when the response type is unrecognized.

**Comment by geauxvirtual** [comment]
snapctl binary needs to be removed from this PR.

File: cmd/snapctl/task.go [hunk 27]
```diff
@@ -48,6 +48,7 @@ var (
 	unionParseFormat = timeParseFormat + " " + dateParseFormat
 )
 
+// Constants K-G are used to truncate task hit and miss counts.
```
Comment (geauxvirtual) [inline cmd/snapctl/task.go]: This should list each constant, and not assume a contiguous list using k-g.

**Comment by geauxvirtual** [comment]
Format of enum comments should be consistent. Some files we comment each enum, in others one comment addresses all the enums.

File: control/available_plugin.go [hunk 28]
```diff
@@ -380,10 +383,10 @@ func (p *apPool) subscribe(taskId string, subType subscriptionType) {
 
 // subscribe adds a subscription to the pool.
```
Comment (geauxvirtual) [inline control/available_plugin.go]: This comment should be updated to say unsubscribe remove a subscription from the pool. This looks like a copy paste error whenever someone added the comments for subscribe and unsubscribe.

File: control/control.go [hunk 29]
```diff
@@ -57,8 +58,9 @@ var (
 	controlLogger = log.WithFields(log.Fields{
 		"_module": "control",
 	})
-
+	// ErrLoadedPluginNotFound - error message for a plugin is not found.
```
Comment (geauxvirtual) [inline control/control.go]: Error message when a plugin is not found.

File: control/control.go [hunk 30]
```diff
@@ -57,8 +58,9 @@ var (
...
 	ErrLoadedPluginNotFound = errors.New("Loaded plugin not found")
+	// ErrControllerNotStarted - error message for a controller is not started error.
```
Comment (geauxvirtual) [inline control/control.go]: Error message when controller is not started.

File: control/control.go [hunk 31]
```diff
@@ -125,21 +127,27 @@ type managesSigning interface {
 	ValidateSignature([]string, string, []byte) error
 }
 
-type ControlOpt func(*pluginControl)
+// PluginControlOpt is the type of pluginControl struture.
+type PluginControlOpt func(*pluginControl)
```
Comment (geauxvirtual) [inline control/control.go]: Why was ControlOpt changed to PluginControlOpt?

File: control/plugin_manager.go [hunk 32]
```diff
@@ -169,51 +175,53 @@ type loadedPlugin struct {
 	ConfigPolicy *cpolicy.ConfigPolicy
 }
 
-// returns plugin name
+// Name eturns plugin name and
```
Comment (geauxvirtual) [inline control/plugin_manager.go]: returns

File: control/control.go (hunk 31 above)
Comment (candysmurf) [inline control/control.go]: Golint does like the name of the variable as the variable is used in the other package as control.ControlOpt, It's stutter to Golint. If you have a better naming option, please share.

**Comment by danielscottt** [comment]
I'm closing this one now that we have smaller, broken-up PRs open.

**Pull request closed without merging**